### DISCUSSION TITLE
maintenanceTimeWindows clarification

### DISCUSSION THREAD

**Discussion by vutkin** [strimzi/strimzi-kafka-operator#5904] [Q&A]
Hi @scholzj , as I correctly understand maintenanceTimeWindows only protects cluster from certificate related activities, but ignore regular reconciliation operations like kafka config change for ex.? Am I right?
For ex. if I set kafka config:
```
"log.message.format.version": "2.7"
```
Operator will start restart of brokers in next reconciliation cycle doesn't respecting maintenanceTimeWindows config.

**Comment by scholzj** [comment]
Yes, correct. The maintenance time window is applied only to the certificate renewal. For other changes, it is not used. The logic is that it is applied for the things which are _automatic_ => that right now is only the certificate renewal. 

The others things are always reactive and are based on some trigger from outside ... for example in this case you trigger it by changing the config. So the operator assumes that you wanna do it now if you changed it now. In other cases it might react to some environment changes ... e.g. when the load-balancer address changes, it needs to react to it otherwise the cluster might not be accessible. So although it might not have been some intentional user action, it still just reacts to some impulse from outside.
